src/Symfony/Component/Serializer/Context/ChildContextFactory.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Context;
+
+use Symfony\Component\Serializer\Normalizer\AbstractObjectNormalizer;
+
+/**
+ * @author Baptiste Leduc <baptiste.leduc@gmail.com>
+ */
+class ChildContextFactory
+{
+    public const ATTRIBUTES = AbstractObjectNormalizer::ATTRIBUTES;
+    public const EXCLUDE_FROM_CACHE_KEY = AbstractObjectNormalizer::EXCLUDE_FROM_CACHE_KEY;
+    public const IGNORED_ATTRIBUTES = AbstractObjectNormalizer::IGNORED_ATTRIBUTES;
+
+    public static function create(array $parentContext, string $attribute, ?string $format = null, array $defaultContext = []): array
+    {
+        if (isset($parentContext[self::ATTRIBUTES][$attribute])) {
+            $parentContext[self::ATTRIBUTES] = $parentContext[self::ATTRIBUTES][$attribute];
+        } else {
+            unset($parentContext[self::ATTRIBUTES]);
+        }
+
+        return $parentContext;
+    }
+
+    public static function createForObject(array $parentContext, string $attribute, ?string $format = null, array $defaultContext = []): array
+    {
+        $parentContext = self::create($parentContext, $attribute, $format, $defaultContext);
+        $parentContext['cache_key'] = self::getAttributesCacheKey($parentContext, $format, $defaultContext);
+
+        return $parentContext;
+    }
+
+    /**
+     * Builds the cache key for the attributes cache.
+     *
+     * The key must be different for every option in the context that could change which attributes should be handled.
+     *
+     * @return bool|string
+     */
+    private static function getAttributesCacheKey(array $context, ?string $format = null, array $defaultContext = [])
+    {
+        foreach ($context[self::EXCLUDE_FROM_CACHE_KEY] ?? $defaultContext[self::EXCLUDE_FROM_CACHE_KEY] ?? [] as $key) {
+            unset($context[$key]);
+        }
+        unset($context[self::EXCLUDE_FROM_CACHE_KEY]);
+        unset($context['cache_key']); // avoid artificially different keys
+
+        try {
+            return md5($format.serialize([
```

Comment:
I don't know if symfony has a policy or a convention about this, but I think all global namespace function should be either wrote with explicit namespace separator such as `\md5()` or be in `use` statements. What's your opinion about this?

Reply:
See https://github.com/symfony/symfony/pull/30956#discussion_r379337597